### PR TITLE
feat(pci.failover-ip): add ipfo order

### PR DESCRIPTION
**Pull request by marie-j**
MSB-90

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.controller.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  $onInit() {
+    this.ip = {
+      quantity: 1,
+      instance: null,
+      region: null,
+      product: null,
+    };
+
+    this.REGIONS = get(find(get(head(this.products), 'details.product.configurations'), { name: 'country' }), 'values');
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.controller.js]: Creating some `const` assignments will improve code readability in this case I think.

It also prevent to use some nested Lodash method calls (four level in this example).

Is that make sense to you?

Example:
```js
const [firstProduct] = this.products;
const configurations = get(firstProduct, 'details.product.configurations')

this.REGIONS = get(find(configurations, { name: 'country' }), 'values');
```

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "pci_projects_project_failoverip_order_quantity": "Nombre d'IP Failover",
+  "pci_projects_project_failoverip_order_instance": "Instance à associer",
+  "pci_projects_project_failoverip_order_region": "Pays de l'IP Failover",
+  "pci_projects_project_failoverip_order_region_invalid": "Votre instance n'est pas compatible avec ce pays",
+  "pci_projects_project_failoverip_order_contracts": "Contrats",
+  "pci_projects_project_failoverip_order_contracts_agree": "J'ai lu et j'accepte les contracts",
+  "pci_projects_project_failoverip_order_check_error": "Une erreur est survenue lors de la vérification de la compatibilité {{ message }}",
+  "pci_projects_project_failoverip_order_success": "Le bon de commande pour votre achat d'IP Failover Cloud est disponible <a href=\"{{ url }}\" target=\"_blank\">ici </a>",
+  "pci_projects_project_failoverip_order_error": "Une erreur est survenue lors de la commande de votre IP Failover {{ message }}",
+  "pci_projects_project_failoverip_order_country_US": "Etats-Unis"
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/translations/Messages_fr_FR.json]:
```suggestion
  "pci_projects_project_failoverip_order_country_US": "États-Unis"
```

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<oui-modal
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.html]: In `packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/order.html` template the `<oui-modal>` is wrap with a `<form>` element.

Shouldn't be the same from this template?

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/order.html [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                data-match="name"
+                required
+                data-on-change="$ctrl.onRegionChange(modelValue)">
+                <span data-ng-bind="$item.name"></span>
+            </oui-select>
+            <div class="oui-field__error" data-ng-class="{
+                'd-block': $ctrl.regionIsInvalid
+            }">
+                <span role="alert">
+                <span class="oui-icon oui-icon-error" aria-hidden="true"></span>
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/order.html]: Missing indentation.

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/translations/Messages_fr_FR.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  "pci_projects_project_failoverip_order_legacy_price": "({{ price }} HT)",
+  "pci_projects_project_failoverip_order_legacy_fees": "Frais d'activation - Pas de renouvellement mensuel",
+  "pci_projects_project_failoverip_order_legacy_country_DE": "Allemagne",
+  "pci_projects_project_failoverip_order_legacy_country_ES": "Espagne",
+  "pci_projects_project_failoverip_order_legacy_country_FR": "France",
+  "pci_projects_project_failoverip_order_legacy_country_PL": "Pologne",
+  "pci_projects_project_failoverip_order_legacy_country_UK": "Royaume-Uni",
+  "pci_projects_project_failoverip_order_legacy_country_IT": "Italie",
+  "pci_projects_project_failoverip_order_legacy_country_CA": "Canada",
+  "pci_projects_project_failoverip_order_legacy_country_US": "Etats-Unis",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/translations/Messages_fr_FR.json]:
```suggestion
  "pci_projects_project_failoverip_order_legacy_country_US": "États-Unis",
```

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+  "pci_projects_project_failoverip_order_legacy": "Acheter des IPS Failover Cloud",
+  "pci_projects_project_failoverip_order_legacy_generate": "Générer le bon de commande",
+  "pci_projects_project_failoverip_order_legacy_cancel": "Annuler",
+  "pci_projects_project_failoverip_order_legacy_description": "Les adresses IPs Failover sont déplaçables d'une instance à l'autre au sein d'un même projet Public Cloud et d'un même continent.",
+  "pci_projects_project_failoverip_order_legacy_quantity": "Sélectionnez le nombre d'IP Failover Cloud que vous souhaitez commander",
+  "pci_projects_project_failoverip_order_legacy_instance": "Sélectionnez l'instance sur laquelle l'IP Failover sera connectée (Cette IP Failover sera prochainement routable vers les autres produits OVH)",
+  "pci_projects_project_failoverip_order_legacy_region": "Sélectionnez le pays de votre IP Failover",
+  "pci_projects_project_failoverip_order_legacy_region_invalid": "Votre instance n'est pas compatible avec ce pays",
+  "pci_projects_project_failoverip_order_legacy_contracts_agree": "J'ai lu et j'accepte les contracts",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/translations/Messages_fr_FR.json]:
```suggestion
  "pci_projects_project_failoverip_order_legacy_contracts_agree": "J'ai lu et j'accepte les contrats",
```

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/translations/Messages_fr_FR.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  "pci_projects_project_failoverip_order": "Acheter des IPS Failover Cloud",
+  "pci_projects_project_failoverip_order_generate": "Générer le bon de commande",
+  "pci_projects_project_failoverip_order_cancel": "Annuler",
+  "pci_projects_project_failoverip_order_description": "Les adresses IPs Failover Cloud sont déplaçables d'un serveur à l'autre au sein d'un même projet Cloud et d'un même continent.",
+  "pci_projects_project_failoverip_order_quantity": "Nombre d'IP Failover",
+  "pci_projects_project_failoverip_order_instance": "Instance à associer",
+  "pci_projects_project_failoverip_order_region": "Pays de l'IP Failover",
+  "pci_projects_project_failoverip_order_region_invalid": "Votre instance n'est pas compatible avec ce pays",
+  "pci_projects_project_failoverip_order_contracts": "Contrats",
+  "pci_projects_project_failoverip_order_contracts_agree": "J'ai lu et j'accepte les contracts",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/translations/Messages_fr_FR.json]:
```suggestion
  "pci_projects_project_failoverip_order_contracts_agree": "J'ai lu et j'accepte les contrats",
```

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/order.routing.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+            projectId,
+          ) => OvhApiCloudProject
+            .Instance()
+            .v6()
+            .query({
+              serviceName: projectId,
+            })
+            .$promise,
+
+          goBack: goToFailoverIps => goToFailoverIps,
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/order.routing.js]:
```suggestion
          goBack: /* @ngInject */ goToFailoverIps => goToFailoverIps,
```

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.html (hunk 3 above)
Comment (marie-j) [reply packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.html]: Indeed

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/order.controller.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
+import get from 'lodash/get';
+
+import { INVALID_COUNTRY_ERROR, REGIONS } from './order.constants';
+// const MESSAGES_CONTAINER_NAME = 'pci.projects.project.failover-ips';
```
Comment (cbourgois) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/legacy/order.controller.js]: Should be removed.

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.controller.js [hunk 10]
```diff
@@ -0,0 +1,61 @@
+import JSURL from 'jsurl';
+import filter from 'lodash/filter';
+import find from 'lodash/find';
+import get from 'lodash/get';
+import isNumber from 'lodash/isNumber';
+import min from 'lodash/min';
+import { ORDER_URL } from './order.constants';
+// const MESSAGES_CONTAINER_NAME = 'pci.projects.project.failover-ips';
```
Comment (cbourgois) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.controller.js]: Should be removed.

File: packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.controller.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    };
+
+    const [product] = this.products;
+    const configurations = get(product, 'details.product.configurations');
+
+    this.REGIONS = get(find(configurations, { name: 'country' }), 'values');
+  }
+
+  static getMaximumQuantity(product) {
+    get(
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/failover-ips/order/agora/order.controller.js]: What do you think about creation multiple `const` assignments in order to keep readability and prevent nested too much Lodash methods.

**Comment by ovh-cds** [comment]
CDS Report public-cloud#2549.0 ■   
* Stage 1
  * Build CA ✔ 
  * Build EU ✔ 
  * Build US ■

**Comment by ovh-cds** [comment]
CDS Report freefax#2549.0 ■   
* Stage 1
  * Build CA ■   
  * Build EU ✔ 
  * Build US ✔

**Comment by antleblanc** [comment]
@marie-j Is there any internal ticket reference related to this pull request?

Thanks!

**Comment by marie-j** [comment]
> @marie-j Is there any internal ticket reference related to this pull request?
> 
> Thanks!

@antleblanc Description has been updated

**Pull request merged**